db/compaction/compaction_service_job.cc

```diff
@@ -221,15 +221,26 @@ CompactionJob::ProcessKeyValueCompactionWithCompactionService(
     }
 
     FileMetaData meta;
+    uint64_t file_size = file.file_size;
+
+    // TODO - Clean this up in the next release.
+    // For backward compatibility - in case the remote worker does not populate
+    // the file_size yet. If missing, continue to populate this from the file
+    // system.
+    if (file_size == 0) {
+      s = fs_->GetFileSize(tgt_file, IOOptions(), &file_size, nullptr);
```

Comment:
nit:maybe add a warning logging for this, since we want to see this usage eventually disappear. It would be nice to confirm it with some logging.